crates/api_common/src/plugins.rs

```diff
@@ -15,18 +15,35 @@ use std::{
   thread::available_parallelism,
   time::Duration,
 };
-use tracing::{debug, warn};
+use tracing::warn;
 
 const GET_PLUGIN_TIMEOUT: Duration = Duration::from_secs(1);
 
 /// Call a plugin hook without rewriting data
 pub fn plugin_hook<T>(name: &'static str, data: &T) -> LemmyResult<()>
 where
-  T: Clone + Serialize + for<'a> Deserialize<'a>,
+  T: Clone + Serialize + for<'b> Deserialize<'b>,
 {
+  let plugins = LemmyPlugins::init();
+  if !plugins.loaded(name) {
+    return Ok(());
+  }
+
   // TODO: as this doesnt return any data it could run multiple plugins in parallel
   //       and/or in background thread
-  Plugins::get().call(name, data)?;
+  for p in plugins.0 {
+    // TODO: add helper method (requires PoolPlugin to be public)
+    // https://github.com/extism/extism/pull/696/files#r2003467812
+    let p = p
+      .plugin_pool
+      .get(&(), GET_PLUGIN_TIMEOUT)?
+      .ok_or(anyhow!("plugin timeout"))?;
+    if p.plugin().function_exists(name) {
+      let params: Json<T> = data.clone().into();
+      p.call::<Json<T>, ()>(name, params)
+        .map_err(|e| LemmyErrorType::PluginError(e.to_string()))?;
+    }
+  }
```

Comment:
Could you externalize this to a function?

Reply:
Unfortunately not, because the `PoolPlugin` type is not public (thats what the todo is for, I left a review comment in the upstream PR so it will hopefully get fixed later). The plugin call can also not be externalized, because the pre hook needs to return data but the post hook not.